lib/populator.js

```diff
@@ -267,12 +276,68 @@ Populator.prototype._enrollInProgram = function (knownKeys, trackedEntityInstanc
   }.bind(this));
 };
 
-Populator.prototype._checkForDuplicateEvent = function (knownKeys, trackedEntityInstanceID, duplicate, next) {
+Populator.prototype._checkForDuplicateEventUsingDataElementID = function (dataElements, trackedEntityInstanceID, duplicate, next) {
+  if (typeof duplicate === 'function') {
+    next = duplicate;
+    duplicate = false;
+  }
+  var uniqueEventDataElementID = this._cache.uniqueEventDataElementID
+  if (!dataElements[uniqueEventDataElementID]) {
+    return next(
+      new Error('No Data Element with ID: ' + uniqueEventDataElementID)
```

Comment:
`ID` should be `UID`

Reply:
This one is actually just an DHIS2 ID. The value of the field is a UID. Though the name is an ID... So it should probably be:
```suggestion
      new Error('No Data Element: ' + uniqueEventDataElementID)
```

Reply:
Cool that sounds good to me :+1: